### PR TITLE
[stable/mongodb-replicaset] Fix tests for cluster with auth

### LINKED ISSUE #9858: [stable/mongodb-replicaset] helm tests are not correct
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: Not relevant

**Which chart**: stable/monodb-replicaset

**What happened**: 
With `auth` turned on we saw the following error when running `helm test`:
```
kubectl logs db-projects-rs0-test --namespace mongodb
1..4
ok 1 Testing mongodb client is accessible
ok 2 Connect mongodb client to mongodb pods
ok 3 Write key to master
not ok 4 Read key from slaves
```

After running several of these tests manually we concluded that tests 2-4 should have failed and test 1 was not very useful.

The core reason these fail is that `MONGOARGS` does not include auth credentials. The reason we get false positives is that no command executes with `--quiet`.

Test 1 should be renamed to `testing mongodb client is executable` to be more clear.

[This test](https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/tests/mongodb-up-test.sh#L50) checks for the word `ok` in the response which is included whether the action succeeds or fails.

[This test](https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/tests/mongodb-up-test.sh#L59) fails because without `--quiet` the output includes the connection port number which is by default `27017`.

**What you expected to happen**:

First, tests 2-4 should fail when `auth` is turned on.
Secondly, tests 2-4 should `auth` when `auth` is turned on, and then pass.

**How to reproduce it** (as minimally and precisely as possible):
`helm test` on a cluster using `auth`.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by tetianakravchenko**
#### What this PR does / why we need it: 
Fix tests to support authentication; minor test fixes: add `--quiet`, etc. 

#### Which issue this PR fixes
  - fixes https://github.com/helm/charts/issues/9858

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/mongodb-replicaset/tests/mongodb-up-test.sh [hunk 1]
```diff
@@ -7,6 +7,10 @@ if [ -f "$MONGOPEM" ]; then
     MONGOARGS="--ssl --sslCAFile $MONGOCACRT --sslPEMKeyFile $MONGOPEM"
 fi
 
+if [ "$AUTH" ]; then
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/tests/mongodb-up-test.sh]: `if [[ "${AUTH}" == "true" ]]` like in [on-start.sh](https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/init/on-start.sh)

File: stable/mongodb-replicaset/tests/mongodb-up-test.sh [hunk 2]
```diff
@@ -34,30 +38,30 @@ setup() {
         sleep 1
 
         for ((i = 0; i < $(replicas); ++i)); do
-            response=$(mongo "$MONGOARGS" "--host=$(pod_name "$i")" "--eval=rs.status()" || true)
-            if [[ "$response" =~ .*ok.* ]]; then
+            response=$(mongo $MONGOARGS "--host=$(pod_name "$i")" "--eval=rs.status()" || true)
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/tests/mongodb-up-test.sh]: `--eval=rs.status().ok` then you can just compare it to "1". Is there any reason we can't remove the regex comparison and just use an equality comparison now?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/mongodb-replicaset/tests/mongodb-up-test.sh (hunk 1 above)
Comment (tetianakravchenko) [reply stable/mongodb-replicaset/tests/mongodb-up-test.sh]: done https://github.com/helm/charts/pull/9907/commits/2da4e991fad32c9400a6d6c4d881d461da6f6a82

File: stable/mongodb-replicaset/tests/mongodb-up-test.sh (hunk 2 above)
Comment (tetianakravchenko) [reply stable/mongodb-replicaset/tests/mongodb-up-test.sh]: good idea, I think it should work. Without `--quiet` this check failed with `1: syntax error: invalid arithmetic operator` error, I've added `--quiet` as a general flag and changed if condition to use equality comparison https://github.com/helm/charts/pull/9907/commits/0a547735bc5689625628fb6f5dfe980275d13f85

**Pull request merged**